src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/Builder.cs

```diff
@@ -220,18 +220,42 @@ protected bool BuildInternal (string projectOrSolution, string target, string []
 
 			LastBuildTime = DateTime.UtcNow - start;
 
-			LastBuildOutput = psi.FileName + " " + args.ToString () + Environment.NewLine;
+			var sb = new StringBuilder ();
+			sb.AppendLine ( psi.FileName + " " + args.ToString () + Environment.NewLine);
 			if (!ranToCompletion)
-				LastBuildOutput += "Build Timed Out!";
-			if (buildLogFullPath != null && File.Exists (buildLogFullPath))
-				LastBuildOutput += File.ReadAllText (buildLogFullPath);
-			LastBuildOutput += string.Format ("\n#stdout begin\n{0}\n#stdout end\n", p.StandardOutput.ReadToEnd ());
-			LastBuildOutput += string.Format ("\n#stderr begin\n{0}\n#stderr end\n", p.StandardError.ReadToEnd ());
-
-			var match = timeElapsedRegEx.Match (LastBuildOutput);
-			if (match.Success) {
-				LastBuildTime = TimeSpan.Parse (match.Groups ["TimeSpan"].Value);
+				sb.AppendLine ("Build Timed Out!");
+			if (buildLogFullPath != null && File.Exists (buildLogFullPath)) {
+				using (var fs = File.OpenRead (buildLogFullPath)) {
+					using (var sr = new StreamReader (fs, Encoding.UTF8, true, 65536)) {
+						string line;
+						while ((line = sr.ReadLine ()) != null) {
+							sb.AppendLine (line);
+						}
+					}
+				}
 			}
+			sb.AppendFormat ("\n#stdout begin\n{0}\n#stdout end\n", p.StandardOutput.ReadToEnd ());
+			sb.AppendFormat ("\n#stderr begin\n{0}\n#stderr end\n", p.StandardError.ReadToEnd ());
+
+			Console.WriteLine ("Searching for Time Elapsed");
+			using (var fs = File.OpenRead (buildLogFullPath)) {
+				var data = new byte [1024];
+				fs.Seek (data.Length, SeekOrigin.End);
+				while (fs.Position > 0) {
```

Comment:
Once concern here; assume a file that is < 1KB in size. The preceding `fs.Seek(data.Length, SeekOrigin.End)` *will throw* an `ArgumentException` (doh!), so it should *probably* instead be:

```csharp
fs.Seek (Math.Min (data.Length, fs.Length), SeekOrigin.End);
```

Additionally, *if*/when that happens, `fs.Position` will be `0` (beginning of file), which means `while (fs.Position > 0)` will *skip over the file contents* and nothing will be parsed.

So long as the file is > 1KB in size, we're fine, but if it isn't...

Reply:
Changed to a do while. and changed the seek too.